misk/src/test/kotlin/misk/web/CorsFilterTest.kt

```diff
@@ -0,0 +1,109 @@
+package misk.web
+
+import com.squareup.moshi.Moshi
+import misk.inject.KAbstractModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.actions.WebAction
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import okhttp3.internal.EMPTY_REQUEST
+import okio.Buffer
+import okio.GzipSink
+import okio.buffer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class CorsFilterTest {
+  @MiskTestModule val module = TestModule()
+
+  @Inject lateinit var jetty: JettyService
+  @Inject lateinit var moshi: Moshi
+
+  @Test fun `Preflight with cors allowed origin`() {
+    val response = preflight("/miskhype/16")
+
+    assertThat(response.headers.get("Access-Control-Allow-Origin"))
+        .isEqualTo("https://localhost:8080")
+  }
+
+  @Test fun `Preflight with cors to restricted domain`() {
```

Comment:
what makes it restricted?

Reply:
I'll rename this test, the endpoint called has a more restrictive policy defined that only allows localhost to make cross domain requests - not that the domain itself is restricted.